src/RouteService/RouteService.Application/Features/Routes/Queries/GetAvailableRoutes/GetAvailableRoutesQueryHandler.cs

```diff
@@ -0,0 +1,28 @@
+﻿using MediatR;
+using RouteService.Application.Exceptions;
+using RouteService.Domain.Entities;
+using RouteService.Domain.Interfaces.Repositories;
+
+namespace RouteService.Application.Features.Routes.Queries.GetAvailableRoutes
+{
+    public class GetAvailableRoutesQueryHandler : IRequestHandler<GetAvailableRoutesQuery, IEnumerable<Ride>?>
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        public GetAvailableRoutesQueryHandler(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task<IEnumerable<Ride>?> Handle(GetAvailableRoutesQuery request, CancellationToken cancellationToken)
+        {
+            var validator = new GetAvailableRoutesQueryValidator();
+            var validationResult = await validator.ValidateAsync(request);
+
+            if (validationResult.Errors.Count > 0)
+                throw new ValidationException(validationResult);
+
+            return await _unitOfWork.RouteRepository.GetAvailableRoutesByQueryAsync(request.From, request.To, request.DepartureTime, request.NumberOfSeats);
```

Comment:
Please set up a mapping between Ride and RideDto, don't return the entity type right to the controller
In RideDto you can put string ExtraInfo property, not including the whole Route object